### DISCUSSION TITLE
getSession serverside don't trigger visibilityChange event to refresh token

### DISCUSSION THREAD

**Discussion by killjoy2013** [nextauthjs/next-auth#4038] [Help]
### Question 💬

Hi,
Using `"next-auth": "^4.2.1",` with CredentialsProvider. Already implemented token rotation and is working alright. On the client side, useSession hook refreshes the session token when it expires. I need to obtain the token to use in my GraphQL query in **getServerSideProps** like this;

```js
  const session = await getSession({
    req,
    broadcast: true,
    triggerEvent: true,
    event: 'visibilitychange',
  });

  if (!session) {
    return {
      redirect: {
        destination: '/login',
        permenant: false,
      },
    };
  }

  const token = await getToken({
    req,
    secret: process.env.TOKEN_SECRET,
    raw: true,
  });
```

As long as I keep the same browser open, this works fine. However, If I close the browser, and come back after token **expiresIn** period, open a new browser and navigate directly to my page `http://localhost:3000/cities`, old token, wich was in the session cookie when I've closed the browser, returns from `getToken`. I'm expecting calling `getSession` with `trigger` & `event` parameters would force jwt process and creates a new token, as in client side.  Am I missing anything? What are the purpose of **event**, **triggerEvent** & **broadcast**  parameters then?  
Thanks in advance

### How to reproduce ☕️

use the code below in getServerSideProps. Navigate to your page directly, like `http://localhost:3000/cities` after token life time.

```js
  const session = await getSession({
    req,
    broadcast: true,
    triggerEvent: true,
    event: 'visibilitychange',
  });

  if (!session) {
    return {
      redirect: {
        destination: '/login',
        permenant: false,
      },
    };
  }

  const token = await getToken({
    req,
    secret: process.env.TOKEN_SECRET,
    raw: true,
  });
```

### Contributing 🙌🏽

No, I am afraid I cannot help regarding this

**Comment by killjoy2013** [comment]
Hi @balazsorban44, 
I didn't open this as bug. However, do you think that it can be?

**Comment by hannanstd** [comment]
I have this problem too

https://github.com/nextauthjs/next-auth/issues/4141

**Reply by killjoy2013** [reply]
just use `getServerSession(ctx)`,  cookie will be updated.
